### DISCUSSION TITLE
0x69B1 is a string !

### DISCUSSION THREAD

**Discussion by afgcons** [stedolan/jq#3052] [Q&A] [answered]
Newbie in distress... After exhausting all my single/double quote back-slash skills I tried this:

```
q='.ZbLight.''0x69B1''.Dimmer';   echo $q
.ZbLight.0x69B1.Dimmer
wget -O - -q "$url2" | jq "$q"
jq: error: syntax error, unexpected LITERAL, expecting $end (Unix shell quoting issues?) at <top-level>, line 1:
.ZbLight.0x69B1.Dimmer
jq: 1 compile error
```

How can I tell jq that 0x69B1 is a string ?

{"ZbLight":{"0x69B1":{"Device":"0x69B1","Na...

**Comment by emanuele6** [comment]
You use `""`:

```sh
wget -qO- -- "$url2" | jq '.ZbLight."0x69B1".Dimmer'
```
(or in your `$q` variable)

**Reply by emanuele6** [reply]
You used `''` (two single quotes) instead of `"` (one double quote) in your code; I edited your comment to use a monospace font to let you see it more clearly.

**Reply by afgcons** [reply]
I replied to your email:

NO GO ...
wget -O - -q "$url2" | jq  '.ZbLight."0x69B1".Dimmer'
jq: error (at <stdin>:0): Cannot index string with string "0x69B1"

{"ZbLight":{"0x69B1":{"Device":"0x69B1","Name":"69B1_CredenzaN","Reachable":true,"Power":1,"Dimmer":50,"Hue":113,"Sat":254,"X":10040,"Y":3116,"CT":0,"ColorMode":2,"RGB":"1FFF00","RGBb":"063200","Light":-1}},"ZbLight":"Done"}

Thanks for helping!

**Reply by wader** [reply, marked as answer]
@afgcons There are two "ZbLight" keys in the root object, second one is the string "Done". I guess that is the reason.
```sh
echo '{"ZbLight":{"0x69B1":{"Device":"0x69B1","Name":"69B1_CredenzaN","Reachable":true,"Power":1,"Dimmer":50,"Hue":113,"Sat":254,"X":10040,"Y":3116,"CT":0,"ColorMode":2,"RGB":"1FFF00","RGBb":"063200","Light":-1}},"ZbLight":"Done"}'  |  jq '.ZbLight'
"Done"
```

**Reply by itchyny** [reply]
Since `.ZbLight` is `"Done"`, you cannot index that value with `."0x69B1"`.

**Reply by afgcons** [reply]
@itchyny - I don't know if this is a json error. It looks "valid". I edited the saved json and re-ran jq after removing the trailing "ZbLight":"Done". It worked... Frustrated but very grateful for helping me, 
@emanuele6 @wader and you: THANK YOU !
